moveos/moveos-types/src/move_std/string.rs

```diff
@@ -35,11 +35,14 @@ impl std::fmt::Debug for MoveString {
 
 impl std::fmt::Display for MoveString {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(
-            f,
-            "{}",
-            std::str::from_utf8(&self.bytes).map_err(|_| std::fmt::Error)?
-        )
+        match std::str::from_utf8(&self.bytes) {
+            Ok(s) => {
+                write!(f, "{}", s)
+            }
+            Err(_) => {
```

Comment:
MoveString is new with utf8 check in the contract, and the Rust implementation should also check utf8

Reply:
If MoveString is a transaction argument or directly deserialized via bcs::from_bcs, we can not check the utf8.